src/HttpHeader.cc

```diff
@@ -664,6 +664,14 @@ HttpHeader::parse(const char *header_start, size_t hdrLen, Http::ContentLengthIn
         }
     }
 
+    if (Config.onoff.ignore_vary) {
+        String rawVary;
+        if (getByIdIfPresent(Http::HdrType::VARY, &rawVary)) {
+            delById(Http::HdrType::VARY);
```

Comment:
AFAICT, this low-level hack will trigger "no leftovers" assertion in Adaptation::Icap::ModXact::prepEchoing() if `ignore_vary` is enabled during reconfiguration that happened after the affected response header has been received but before ICAP decided to echo that response.

Most likely, this code should be moved to (i.e. triggered from) higher level code to avoid such difficult-to-find bugs.